src/ui/router/Browse.js

```diff
@@ -36,10 +38,19 @@ define([
         }
 
         function viewObject(object, viewProvider) {
+            if (mutable) {
+                openmct.objects.destroyMutable(mutable);
+                mutable = undefined;
+            }
+
+            if (openmct.objects.supportsMutation(object)) {
+                mutable = openmct.objects._toMutable(object);
+            }
+
             currentObjectPath = openmct.router.path;
 
-            openmct.layout.$refs.browseObject.show(object, viewProvider.key, true, currentObjectPath);
-            openmct.layout.$refs.browseBar.domainObject = object;
+            openmct.layout.$refs.browseObject.show(mutable || object, viewProvider.key, true, currentObjectPath);
+            openmct.layout.$refs.browseBar.domainObject = mutable || object;
```

Comment:
I know objects truthiness is pretty absolute but shouldn't we be explicitly checking if its defined?